packages/ledgerstate/color.go

```diff
@@ -0,0 +1,228 @@
+package ledgerstate
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/iotaledger/hive.go/datastructure/orderedmap"
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/stringify"
+	"github.com/mr-tron/base58"
+	"golang.org/x/xerrors"
+)
+
+// region Color ////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// ColorLength represents the length of a Color (amount of bytes).
+const ColorLength = 32
+
+// Color represents a marker that is associated to a token balance and that can give tokens a certain "meaning".
+type Color [ColorLength]byte
+
+// ColorFromBytes unmarshals a Color from a sequence of bytes.
+func ColorFromBytes(bytes []byte) (color Color, consumedBytes int, err error) {
+	marshalUtil := marshalutil.New(bytes)
+	if color, err = ColorFromMarshalUtil(marshalUtil); err != nil {
+		err = xerrors.Errorf("failed to parse Color from MarshalUtil: %w", err)
+		return
+	}
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ColorFromBase58EncodedString creates a Color from a base58 encoded string.
+func ColorFromBase58EncodedString(base58String string) (color Color, err error) {
+	bytes, err := base58.Decode(base58String)
+	if err != nil {
+		err = xerrors.Errorf("error while decoding base58 encoded Color (%v): %w", err, ErrBase58DecodeFailed)
+		return
+	}
+
+	if color, _, err = ColorFromBytes(bytes); err != nil {
+		err = xerrors.Errorf("failed to parse Color from bytes: %w", err)
+		return
+	}
+
+	return
+}
+
+// ColorFromMarshalUtil unmarshals a Color using a MarshalUtil (for easier unmarshaling).
+func ColorFromMarshalUtil(marshalUtil *marshalutil.MarshalUtil) (color Color, err error) {
+	colorBytes, err := marshalUtil.ReadBytes(ColorLength)
+	if err != nil {
+		err = xerrors.Errorf("failed to parse Color (%v): %w", err, ErrParseBytesFailed)
+		return
+	}
+	copy(color[:], colorBytes)
+
+	return
+}
+
+// Bytes marshals the Color into a sequence of bytes.
+func (c Color) Bytes() []byte {
+	return c[:]
+}
+
+// Base58 returns a base58 encoded version of the Color.
+func (c Color) Base58() string {
+	return base58.Encode(c.Bytes())
+}
+
+// String creates a human readable string of the Color.
+func (c Color) String() string {
+	switch c {
+	case ColorIOTA:
+		return "IOTA"
+	case ColorMint:
+		return "MINT"
+	default:
+		return c.Base58()
+	}
+}
+
+// ColorIOTA is the zero value of the Color and represents uncolored tokens.
+var ColorIOTA = Color{}
```

Comment:
Why are these Colors at the bottom and not constants?
I'd move them to the top and if no particular reason for being `var` make them `const`.

Reply:
We can move them to the top but Colors cannot be constants because constants in Go can only be scalar types like string, integer and so on.